### DISCUSSION TITLE
Track Editor in GoDot Engine or Ingame?

### DISCUSSION THREAD

**Discussion by Jean28518** [Libre-TrainSim/Libre-TrainSim#50] [Ideas]
# GoDot Engine Editor:
## Positive:
 + It already exists, wo don't have much work developing it further on
 + Complex features like lights, custom animations, are already integrated.
## Negative:
 - The Editor is in some cases very unhandy and not intuitive. Changing some behaviours is very very hard/not possible.
 - The handling of big tracks above 30 kms becomes very unhandy, and laggy

# Ingame Editor:
## Positive:
 + The editor itself could be much easier, and more accessible to newbies. 
 + We can fully decide, how to design the editor and workflow
 + It runs at lower hardware. Even making tracks at mobile phone would be possible eventually
## Negative:
 - It takes much effort and work to develop
 - It only has the features which we implement to it. Custom modifications on a track are not possible.
 - At some points we will have hard work, to make it possible. (For example easy clicking at a single object)

Let's discuss about it!

**Comment by comick** [comment]
In-game track editor could be ok, but having both would possibly be best. e.g.: I'm playing with track that uses a popular godot terrain plugin (https://github.com/Zylann/godot_terrain_plugin) and open elevation data. How would you do that in-game?

Simple in-game is extremely desirable and would lower the threshold for people to contribute and play with own tracks, but in my opinion godot editable tracks should stay for higher quality tracks.

**Comment by Jean28518** [comment]
Thanks for your opinion.
Yeah, I know about Zylanns Plugin, but I think, I will develop a landscape tool for fitted for Libre TrainSim.. 
I am learning about it at university this month..

**Comment by bodamat** [comment]
Great university! I want to study it because I learn only Pascal in it.
I think for now Godot editor enough, but in the future, both editors are better because Godot has more future for creating, grow up every day, and doesn't need to do something, but the in-game editor needs to do something every day and waste time :( Maybe the in-game editor in the future won't need it.
